portable/GCC/ARM_CM0/portasm.c

```diff
@@ -443,49 +443,52 @@ void vClearInterruptMask( __attribute__( ( unused ) ) uint32_t ulMask ) /* __att
-            "     cmp r2, r3                 \n"
-            "     beq system_call_exit       \n"
-            "     b vPortSVCHandler_C        \n"
-            "                                \n"
-            " system_call_enter:             \n"
-            "    push {lr}                   \n"
-            "    bl vSystemCallEnter         \n"
-            "    pop {pc}                    \n"
-            "                                \n"
-            " system_call_exit:              \n"
-            "    push {lr}                   \n"
-            "    bl vSystemCallExit          \n"
-            "    pop {pc}                    \n"
-            "                                \n"
-            " .align 4                       \n"
-            "                                \n"
+            " .syntax unified                                       \n"
+            " .extern vPortSVCHandler_C                             \n"
+            " .extern vSystemCallEnter                              \n"
+            " .extern vSystemCallExit                               \n"
+            " .extern pxCurrentTCB                                  \n"
+            "                                                       \n"
+            " movs r0, #4                                           \n"
+            " mov r1, lr                                            \n"
+            " tst r0, r1                                            \n"
+            " beq stack_on_msp                                      \n"
+            "                                                       \n"
+            " stack_on_psp:                                         \n"
+            "     mrs r0, psp                                       \n"
+            "     b route_svc                                       \n"
+            "                                                       \n"
+            " stack_on_msp:                                         \n"
+            "     mrs r0, msp                                       \n"
+            "     b route_svc                                       \n"
+            "                                                       \n"
+            " route_svc:                                            \n"
+            "     ldr r3, [r0, #24]                                 \n"
+            "     subs r3, #2                                       \n"
+            "     ldrb r2, [r3, #0]                                 \n"
+            "     ldr r3, =%0                                       \n"
+            "     cmp r2, r3                                        \n"
+            "     blt system_call_enter                             \n"
+            "     ldr r3, =%1                                       \n"
+            "     cmp r2, r3                                        \n"
+            "     beq system_call_exit                              \n"
+            "     ldr r3, vPortSVCHandler_C_Address                 \n"
```

Comment:
```suggestion
            "     ldr r3, = vPortSVCHandler_C                 \n"
```
@TomasGalbickaNXP @aggarg Apologies for the nit. With this style, we can avoid the constant pool defined at the end of the function and let the linker handle the needed alignement.